src/Microsoft.Diagnostics.EventFlow.Inputs.NLog/NLogInput.cs

```diff
@@ -0,0 +1,265 @@
+﻿// ------------------------------------------------------------
+//  Copyright (c) Microsoft Corporation.  All rights reserved.
+//  Licensed under the MIT License (MIT). See License.txt in the repo root for license information.
+// ------------------------------------------------------------
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using Microsoft.Diagnostics.EventFlow.Metadata;
+using NLog;
+using NLog.Config;
+using NLog.Targets;
+
+namespace Microsoft.Diagnostics.EventFlow.Inputs
+{
+    [Target(NLogTargetTypeName)]
+    public class NLogInput : TargetWithContext, IObservable<EventData>
+    {
+        internal const string NLogTargetTypeName = "EventFlowInput";
+
+        private static readonly IDictionary<NLog.LogLevel, LogLevel> ToLogLevel = new Dictionary<NLog.LogLevel, LogLevel>
+        {
+            [NLog.LogLevel.Trace] = LogLevel.Verbose,
+            [NLog.LogLevel.Debug] = LogLevel.Verbose,
+            [NLog.LogLevel.Info] = LogLevel.Informational,
+            [NLog.LogLevel.Warn] = LogLevel.Warning,
+            [NLog.LogLevel.Error] = LogLevel.Error,
+            [NLog.LogLevel.Fatal] = LogLevel.Critical
+        };
+
+        private readonly EventFlowSubject<EventData> _eventFlowSubject;
+        private readonly IHealthReporter _healthReporter;
+
+        /// <inheritdoc/>
+        public override IList<TargetPropertyWithContext> ContextProperties { get; } = new List<TargetPropertyWithContext>();
+
+        public NLogInput(IHealthReporter healthReporter)
+        {
+            _healthReporter = healthReporter;
+            OptimizeBufferReuse = true;
+            IncludeEventProperties = true;
+            Layout = "${message}";
+
+            RegisterNLogTargetType(healthReporter, GetType());
+
+            _eventFlowSubject = new EventFlowSubject<EventData>();
+        }
+
+        protected override void CloseTarget()
+        {
+            _eventFlowSubject?.Dispose();
```

Comment:
Just curious: do we ever expect the `_eventFlowSubject` to be null?

If not, then ordinary "dot" operator for function call is sufficient